src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -63,27 +65,124 @@ public class PermalinkService : IPermalinkService
             _tableBuilderService = tableBuilderService;
             _permalinkCsvMetaService = permalinkCsvMetaService;
             _blobStorageService = blobStorageService;
+            _frontendService = frontendService;
             _subjectRepository = subjectRepository;
             _publicationRepository = publicationRepository;
             _releaseRepository = releaseRepository;
             _mapper = mapper;
         }
 
+        public async Task<Either<ActionResult, PermalinkViewModel>> GetPermalink(Guid permalinkId,
+            CancellationToken cancellationToken = default)
+        {
+            return await Find(permalinkId, cancellationToken).OnSuccess(BuildViewModel);
+        }
+
+        private async Task<Either<ActionResult, Permalink>> Find(Guid permalinkId,
+            CancellationToken cancellationToken)
+        {
+            return await _contentDbContext.Permalinks
+                .SingleOrNotFoundAsync(
```

Comment:
As it seem like we're now storing the entire table inside of the content DB's `Permalink` model, calling `Single` (and similar methods) will fetch the entire table row, including the table.

This means that it's completely redundant to then re-fetch the table from blob storage.

If we want to be more efficient here, we'll have to do one of the following:
- Refactor this to use `AnyAsync` (although should check that this isn't secretly doing a `SELECT *`)
- Just use the table from the DB row as-is. See other comment on having the tables in both the content DB and blob storage.